code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,494 @@
+
+
+/datum/action/xeno_action/activable/blink/proc/blink_warp(mob/living/carbon/xenomorph/wraith/ghost, turf/T, mob/pulled = null) //This handles the location swap between the Wraith and the Warp Shadow
+
+	ghost.face_atom(T) //Face the target so we don't look like an ass
+	ghost.add_filter("wraith_blink_warp_filter", 3, list("type" = "blur", 5)) //Cool filter appear
+	addtimer(CALLBACK(ghost, /atom.proc/remove_filter, "wraith_blink_warp_filter"), 0.5 SECONDS) //0.5 sec blur duration
+	addtimer(VARSET_CALLBACK(ghost, alpha, initial(ghost.alpha)), 0.5 SECONDS)
+
+	teleport_debuff_aoe(ghost) //Debuff when we vanish
+
+	ghost.forceMove(T) //Teleport to our target turf
+
+	if(pulled) //bring the pulled target with us if applicable
+		cooldown_timer *= WRAITH_BLINK_DRAG_MULTIPLIER
+		to_chat(ghost, "<span class='xenodanger'>We bring [pulled.name] with us. We won't be ready to blink again for [cooldown_timer * 0.1] seconds due to the strain of doing so.</span>")
+
+	teleport_debuff_aoe(ghost) //Debuff when we reappear
+
+
+//Called by many of the Wraith's teleportation effects
+/datum/action/xeno_action/proc/teleport_debuff_aoe(atom/movable/teleporter, silent = FALSE, no_visuals = FALSE)
+
+	var/mob/living/carbon/xenomorph/ghost = owner
+
+	if(!silent) //Sound effects
+		playsound(teleporter, 'sound/effects/EMPulse.ogg', 25, 1) //Sound at the location we are arriving at
+
+	if(!no_visuals) //Visuals
+		new /obj/effect/temp_visual/blink_portal(get_turf(teleporter))
+
+
+	for(var/turf/affected_tile in range(1,teleporter.loc))
+		affected_tile.Shake(4, 4, 1 SECONDS) //Cool SFX
+		affected_tile.add_filter("wraith_blink_distortion",1,list("type"="radial_blur", "size" = 0.5))
+		addtimer(CALLBACK(affected_tile, /atom.proc/remove_filter, "wraith_blink_distortion"), 1 SECONDS)
+		for(var/mob/living/target in affected_tile)
+			target.Shake(4, 4, 1 SECONDS) //SFX
+			target.add_filter("wraith_aoe_debuff_filter", 3, list("type" = "blur", 5)) //Cool filter appear
+			addtimer(CALLBACK(target, /atom.proc/remove_filter, "wraith_aoe_debuff_filter"), 1 SECONDS) //1 sec blur duration
+			if(target.stat == DEAD)
+				continue
+			if(isxeno(target))
+				var/mob/living/carbon/xenomorph/X = target
+				if(X.issamexenohive(ghost)) //No friendly fire
+					continue
+
+			shake_camera(target, 2, 1)
+			target.adjust_stagger(WRAITH_TELEPORT_DEBUFF_STACKS)
+			target.add_slowdown(WRAITH_TELEPORT_DEBUFF_STACKS)
+			target.adjust_blurriness(WRAITH_TELEPORT_DEBUFF_STACKS) //minor visual distortion
+			to_chat(target, "<span class='warning'>You feel nauseous as the world warps around you!</span>")
+
+/datum/action/xeno_action/activable/blink/on_cooldown_finish()
+
+	if(cooldown_timer > initial(cooldown_timer) ) //Reset the cooldown if increased from dragging someone.
+		to_chat(owner, "<span class='xenodanger'>We are able to blink again.</span>")
+		owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+		cooldown_timer = initial(cooldown_timer)
```

Comment:
Why

Reply:
The next Blink cooldown is dramatically increased if the Wraith blinks while grabbing someone, as it teleports with that mob; the added cooldown is the price of the additional value/utility; this reverts that increase.

Reply:
Actually, I should probably change add_cooldown so it can accept modifier arguments